VManalyze/vm_analyze.cpp

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string.h> 
+#include "pin.H"
+
+using std::cerr;
+using std::ios;
+using std::string;
+using std::endl;
+
+#define LIBC_BASE 0x700000000000
+
+std::vector<int> switch_jmps;
+std::vector<int> jmp_ins;
+ADDRINT image_base;
+
+VOID printSwitch(ADDRINT ip, CONTEXT *ctxt){ 
+    ADDRINT val;
+    PIN_GetContextRegval(ctxt, LEVEL_BASE::REG_RAX, reinterpret_cast<UINT8*>(&val));
+    val = val - image_base;
+    if (val > 50){
+        switch_jmps.push_back(val);
+    }
+}
+
+VOID Trace(TRACE trace, VOID *v)
+{
+    // Visit every basic block  in the trace
+    INS tail;
+    ADDRINT adr;
+    for(BBL bbl = TRACE_BblHead(trace); BBL_Valid(bbl); bbl = BBL_Next(bbl))
+    {
+        tail = BBL_InsTail(bbl);
+        adr = INS_Address(tail);
+        
+        // Here we fetch every block and disassemble the tail instruction 
+        if((INS_Disassemble(tail).rfind("jmp rax") == 0) && (adr < LIBC_BASE)){
```

Comment:
do switch statements always resolve to `jmp rax`? even if we are being hackish here can't it be any register?

Reply:
I use babyvm challenge as reference. It was jmp rax in both of them.

Reply:
merging this for now, lets figure out a better way to do this down the line